Library/Homebrew/formula.rb

```diff
@@ -3201,6 +3201,7 @@ def skip_clean(*paths)
     # <pre>keg_only :versioned_formulae</pre>
     # <pre>keg_only "because I want it so"</pre>
     def keg_only(reason, explanation = "")
+      reason = reason&.gsub(HOMEBREW_PREFIX, HOMEBREW_PREFIX_PLACEHOLDER) if reason.is_a?(String)
```

Comment:
Ah! 💡 I think I only want to do this replacement if we're installing from the API, since it's only _from_ the API that we overwrite with the real prefix value?